File: Valid Palindrome/Valid_Palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string ans = "";
        int n = s.size();
        for(int i = 0; i<n; i++) {
            char x = tolower(s[i]);
            if('0' <= x && x <= '9') {
                ans += x;
            }
            else if('a' <= x && x <= 'z') {
                ans += x;
            }
        }
        int left = 0,right = ans.size()-1;
        while(left < right) {
            if(ans[left] != ans[right]) {
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
};

File: Valid Palindrome/Valid_Palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        ans = ""
        for char in s.lower():
            if '0' <= char <= '9' or 'a' <= char <= 'z':
                ans += char
        left, right = 0, len(ans) - 1
        while left < right:
            if ans[left] != ans[right]:
                return False
            left += 1
            right -= 1
        return True







        